From d9b8f0295ab5cf47caf35384d33789f372c72a8f Mon Sep 17 00:00:00 2001
From: Stanislaw Kardach <kda@semihalf.com>
Date: Wed, 17 Oct 2018 14:51:15 +0300
Subject: [PATCH 0291/1051] octeontx: only require PKI VF if domain has ports

Change domain requirements so that it does not require PKI VF if there
are no ports specified in the Application Domain specification.

Fix build issues.

Signed-off-by: Stanislaw Kardach <kda@semihalf.com>
Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
Signed-off-by: Yury Norov <ynorov@marvell.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c
index a4eae61f2666..858e303b2f0e 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c
@@ -521,8 +521,8 @@ int octeontx_create_domain(const char *name, int type, int sso_count,
 		return -EINVAL;
 	}
 
-	if (!pki_count) {
-		dev_err(octtx_device, "Domain has to include exactly 1 PKI\n");
+	if ((bgx_count + lbk_count) != 0 && pki_count != 1) {
+		dev_err(octtx_device, "Domain has to include exactly 1 PKI if there are BGX or LBK ports\n");
 		return -EINVAL;
 	}
 
-- 
2.17.1

